webapp/src/services/api/studies/tableMode/constants.ts

```diff
@@ -1,73 +1,121 @@
-const AREA = "area";
-const LINK = "link";
-const CLUSTER = "cluster";
-const RENEWABLE = "renewable";
-const BINDING_CONSTRAINT = "binding constraint";
+const AREA = "areas";
+const LINK = "links";
+const THERMAL = "thermals";
+const RENEWABLE = "renewables";
+// Avoid "storages" because we may have "lt-storages" (long-term storages) in the future
+const ST_STORAGE = "st-storages";
+// Avoid "constraints" because we may have other kinds of constraints in the future
+const BINDING_CONSTRAINT = "binding-constraints";
 
 export const TABLE_MODE_TYPES = [
   AREA,
   LINK,
-  CLUSTER,
+  THERMAL,
   RENEWABLE,
+  ST_STORAGE,
   BINDING_CONSTRAINT,
 ] as const;
 
-export const TABLE_MODE_COLUMNS_BY_TYPE = {
-  [AREA]: [
-    // Optimization - Nodal optimization
-    "nonDispatchablePower",
-    "dispatchableHydroPower",
-    "otherDispatchablePower",
-    "averageUnsuppliedEnergyCost",
-    "spreadUnsuppliedEnergyCost",
-    "averageSpilledEnergyCost",
-    "spreadSpilledEnergyCost",
-    // Optimization - Filtering
-    "filterSynthesis",
-    "filterYearByYear",
-    // Adequacy patch
-    "adequacyPatchMode",
-  ],
-  [LINK]: [
-    "hurdlesCost",
-    "loopFlow",
-    "usePhaseShifter",
-    "transmissionCapacities",
-    "assetType",
-    "linkStyle",
-    "linkWidth",
-    "displayComments",
-    "filterSynthesis",
-    "filterYearByYear",
-  ],
-  [CLUSTER]: [
-    "group",
-    "enabled",
-    "mustRun",
-    "unitCount",
-    "nominalCapacity",
-    "minStablePower",
-    "spinning",
-    "minUpTime",
-    "minDownTime",
-    "co2",
-    "marginalCost",
-    "fixedCost",
-    "startupCost",
-    "marketBidCost",
-    "spreadCost",
-    "tsGen",
-    "volatilityForced",
-    "volatilityPlanned",
-    "lawForced",
-    "lawPlanned",
-  ],
-  [RENEWABLE]: [
-    "group",
-    "tsInterpretation",
-    "enabled",
-    "unitCount",
-    "nominalCapacity",
-  ],
-  [BINDING_CONSTRAINT]: ["type", "operator", "enabled"],
+/**
+ * Mapping of table types to their columns configuration.
+ * The keys are the table types, and the values are objects with keys being the study version since which the columns are available.
+ */
+export const TABLE_MODE_COLUMNS_MAPPING: Readonly<
+  Record<string, Readonly<Record<number, readonly string[]>>>
+> = {
+  [AREA]: {
+    600: [
+      "nonDispatchablePower",
+      "dispatchableHydroPower",
+      "otherDispatchablePower",
+      "averageUnsuppliedEnergyCost",
+      "spreadUnsuppliedEnergyCost",
+      "averageSpilledEnergyCost",
+      "spreadSpilledEnergyCost",
+      "filterSynthesis",
+      "filterYearByYear",
+      "adequacyPatchMode",
+    ],
+    830: ["adequacyPatchMode"],
+  },
```

Comment:
the data structure is tightly coupled with the implementation details, specifically the study versions
this design choice may lead to several issues regarding maintainability and scalability. here's how you can refine types and streamline the data structure to better handle versioning:

types:
```typescript
const Column = {
  Area: "areas",
  Link: "links",
  Thermal: "thermals",
  Renewable: "renewables",
  ShortTermStorage: "st-storages",
  BindingConstraint: "binding-constraints",
} as const;

export type ColumnType = (typeof Column)[keyof typeof Column];
```
structure example:
```typescript
export const TABLE_MODE_COLUMNS = {
  [Column.Area]: {
    base: [
      "nonDispatchablePower",
      "dispatchableHydroPower",
      "otherDispatchablePower",
      "averageUnsuppliedEnergyCost",
      "spreadUnsuppliedEnergyCost",
      "averageSpilledEnergyCost",
      "spreadSpilledEnergyCost",
      "filterSynthesis",
      "filterYearByYear",
    ],
    conditional: [{ columns: ["adequacyPatchMode"], minVersion: 830 }],
  },
  [Column.Link]: {
    base: [
      "hurdlesCost",
      "loopFlow",
      "usePhaseShifter",
      "transmissionCapacities",
      "assetType",
      "linkStyle",
      "linkWidth",
      "comments",
      "displayComments",
      "filterSynthesis",
      "filterYearByYear",
    ],
    conditional: [],
  },
  [Column.Thermal]: {
    base: [
      "group",
      "enabled",
      "unitCount",
      "nominalCapacity",
      "genTs",
      "minStablePower",
      "minUpTime",
      "minDownTime",
      "mustRun",
      "spinning",
      "volatilityForced",
      "volatilityPlanned",
      "lawForced",
      "lawPlanned",
      "marginalCost",
      "spreadCost",
      "fixedCost",
      "startupCost",
      "marketBidCost",
      "co2",
    ],
    conditional: [
      {
        columns: [
          "nh3",
          "so2",
          "nox",
          "pm25",
          "pm5",
          "pm10",
          "nmvoc",
          "op1",
          "op2",
          "op3",
          "op4",
          "op5",
        ],
        minVersion: 860,
      },
      {
        columns: ["costGeneration", "efficiency", "variableOMCost"],
        minVersion: 870,
      },
    ],
  },
etc...
```

Usage:
```typescript
export function getColumnsByType(type: ColumnType, version: number) {
  const { base, conditional } = TABLE_MODE_COLUMNS[type];

  return conditional.reduce(
    (acc, item) => {
      if (version >= item.minVersion) {
        return [...acc, ...item.columns];
      }
      return acc;
    },
    [...base],
  );
}
```

Reply:
Cool

Reply:
its more descriptive I agree, 
``` typescript
const Column = {
  Area: "areas",
  Link: "links",
  Thermal: "thermals",
  Renewable: "renewables",
  ShortTermStorage: "st-storages",
  BindingConstraint: "binding-constraints",
} as const;
```
this pattern is called an "enum-like" object we use this instead of regular TS enums because it's more type safe.

for the rest, the structure you suggest is basically the same as the original one I suggested to change. It's not structured and the use-cases are mixed. Also we don't use OOP in React, but rather the functional paradigm like I explained above.

I will let @skamril review your changes and take the rest, hope my comments are helpful